homeassistant/components/device_tracker/freebox.py

```diff
@@ -1,56 +1,25 @@
 """
-Support for device tracking through Freebox routers.
+Support for Freebox devices (Freebox v6 and Freebox mini 4K).
 
-This tracker keeps track of the devices connected to the configured Freebox.
-
-For more details about this platform, please refer to the documentation at
-https://home-assistant.io/components/device_tracker.freebox/
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/device_tracker/freebox/
 """
-import asyncio
-import copy
-import logging
-import socket
 from collections import namedtuple
-from datetime import timedelta
-
-import voluptuous as vol
+import logging
 
-import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.event import async_track_time_interval
-from homeassistant.components.device_tracker import (
-    PLATFORM_SCHEMA, CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
-from homeassistant.const import (
-    CONF_HOST, CONF_PORT)
+from homeassistant.components.device_tracker import DeviceScanner
+from homeassistant.components.freebox import DATA_FREEBOX
 
-REQUIREMENTS = ['aiofreepybox==0.0.5']
+DEPENDENCIES = ['freebox']
 
 _LOGGER = logging.getLogger(__name__)
 
-FREEBOX_CONFIG_FILE = 'freebox.conf'
-
-PLATFORM_SCHEMA = vol.All(
-    PLATFORM_SCHEMA.extend({
-        vol.Required(CONF_HOST): cv.string,
-        vol.Required(CONF_PORT): cv.port
-    }))
-
-MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
-
-
-async def async_setup_scanner(hass, config, async_see, discovery_info=None):
-    """Set up the Freebox device tracker and start the polling."""
-    freebox_config = copy.deepcopy(config)
-    if discovery_info is not None:
-        freebox_config[CONF_HOST] = discovery_info['properties']['api_domain']
-        freebox_config[CONF_PORT] = discovery_info['properties']['https_port']
-        _LOGGER.info("Discovered Freebox server: %s:%s",
-                     freebox_config[CONF_HOST], freebox_config[CONF_PORT])
-
-    scanner = FreeboxDeviceScanner(hass, freebox_config, async_see)
-    interval = freebox_config.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
-    await scanner.async_start(hass, interval)
-    return True
 
+async def async_get_scanner(hass, config):
+    """Validate the configuration and return a Freebox scanner."""
+    scanner = FreeboxDeviceScanner(hass.data[DATA_FREEBOX])
```

Comment:
Since the freebox component is initialized asynchronously, this triggers a KeyError on startup when the device tracker platform is loaded before the component initialization is finished. I don't know how to fix that.

Reply:
Do you have this KeyError while using discovery or fixed configuration ? Or both ?
I can't reproduce this error at the moment.

Reply:
I have it with fixed configuration, haven't tested with discovery.

Reply:
Did you configured a device_tracker platform ? Because with this new version you have only to configure : 
`freebox:`
`  host: foobar.fbxos.fr`
`  port: 1234`

No more device_tracker with freebox !

Reply:
Oh right, that was the issue! I still had the old config with the device tracker platform, after updating my config file no more error on startup, nice job!